base/genimport/importer.go

```diff
@@ -329,11 +345,14 @@ func goModReplacementDirectives(o *Output) [][]byte {
 	}
 
 	// Replace the current module with the right directory
-	replacementDirectives = append(replacementDirectives, []byte("replace "+moduleName+" => "+cwd+"\n"))
+	replacementDirectives := [][]byte{
+		[]byte("replace "+moduleName+" => "+goModDir+"\n"),
+	}
 
 	// Copy over the replaces, since they only apply when in the main module
 	for _, line := range origLines {
 		// TODO: this doesn't handle block syntax e.g. replace ( ... )
+		// TODO: this doesn't work if the replace directives found are themselves localized
```

Comment:
Ah, great point!  Then we'll definitely need a proper parser so we can extract the trace paths, I think.